### ISSUE TITLE
Decide what numbers are important to track

### ISSUE THREAD

**Issue by adamlofting** [adamlofting/dashboard-zero#1]
I suggest we start by looking at the numbers found in Github. Most Mozilla Foundation engineering work happens in Github, and there is a large source of data we can query and surface on a dashboard.

Here are some of the ideas from the original ticket.
- [ ] Number of unanswered volunteer input in Github issues
- [ ] Number of Pull Requests that have had no response at all
- [ ] Number of Pull Requests that have had no activity for more than X days
- [ ] Number of open bugs (total)
- [ ] Number of bugs needing triage
- [ ] Number of bugs by triage status
- [ ] Number of bugs assigned to expired Milestones
- [ ] Number of Repos with code coverage below X %

Do you like this list? Are there any we should remove? Or any others you think we should add?

**Comment by gngdhr** [comment]
Adam, given that bugzilla is for the issue tracking, are you considering having the bugs (and new requirements too) on bugzilla also listed along with the issues in Git?

**Comment by adamlofting** [comment]
@gngdhr thanks for the note, but actually for the Foundation projects we're looking at (Webmaker, Mozilla Learning, Thimble, Advocacy, and so on) the issue tracking moved to Github in 2015.

**Comment by drazisil** [comment]
I've been looking into the GitHub API recently, and I believe all of these are doable for the most part.

Two notes:
- Number of unanswered volunteer input in Github issues

This metric is comments not from Mozilla team on issues?
- Number of Repos with code coverage below X %

afaik, this is not a Github metric. Where would the code coverage data source be coming from?

**Comment by drazisil** [comment]
@adamlofting I saw that @rjack created https://github.com/rjack/dashboard-zero/blob/my-experiments/my-experiments/github-api/mozilla-repos-count-by-organization.txt which looks really helpful.

Do you have a list of orgs/repos this project would be geared towards, or is that it?

**Comment by drazisil** [comment]
@gsilvapt: I'm using https://github.com/mozilla?utf8=%E2%9C%93&query=+only%3Asources+ as a starting data source.

I'm gathering the following metrics to start:
- repo_name
- number_of_stars
- number of forks
- number_of_open_issues

From there I'm reaching out and getting the following metrics for each issue:
- issue_number
- issue_name
- issue_create_datetimestamp
- is_issue_pull_request

(Item of note, pull requests are issues as well, so they are in the same list.)

Mozilla has 1085 public repos listed that are not forks (forks don't have issues so there are less helpful). Once I have this list I'll share it so we can start running data/hacking on it.

Any suggestions for more useful / additional data welcome!

**Comment by gsilvapt** [comment]
That's a great starting point. In what platform are you doing that? 

Just a small suggestion? Can you have a secondary field somewhere that shows which language is being used on which repo? Perhaps that may be useful to allocate resources.

**Comment by drazisil** [comment]
@gsilvapt 

In nodejs, though the initial data pulls would probably be better in curl

I like the idea of language, that would be a good metric to keep track of as well.

**Comment by drazisil** [comment]
Here's the update with language: https://github.com/drazisil/dashboard-zero/blob/jucy-stats/data/repos.csv

**Comment by adamlofting** [comment]
@drazisil 

> This metric is comments not from Mozilla team on issues?

Yes, I would use the 'Members' API to see who has edit rights on a repository, and treat those users as the 'Mozilla team'. That's not always staff, and it's not all staff either, as staff often contribute to other team projects in their free time. But I think it's a reasonable proxy. https://developer.github.com/v3/orgs/members/

>  Number of Repos with code coverage below X %
> afaik, this is not a Github metric. Where would the code coverage data source be coming from?

Yes, let's skip this for now and stick to the Github data. We were talking about adding something like https://coveralls.io to the repos we were tracking, but that's not consistently available yet. 

> Mozilla has 1085 public repos listed that are not forks (forks don't have issues so there are less helpful). Once I have this list I'll share it so we can start running data/hacking on it.

In practice, many people will only work on a small subset of those repos, so we'll want to be able to setup dashboards for an individual team. It's only going to be a useful dashboard if the people looking at it are able to take action to 'zero' the numbers they are looking at. So, for the sake of designing this, we can build a prototype using let's say, five repos that are currently active.

**Comment by adamlofting** [comment]
In case it's useful, this repo includes a fair number of Nodejs exaples for scraping data from the Github API: https://github.com/adamlofting/gitribution-2

**Comment by drazisil** [comment]
@adamlofting i can absolutly set it so you can build a 'team' of repos vs all of them, that will make staying in the rate limits a lot easier as well. 

The members list was where i was planning on going, unless you had another plan :)

Regarding the scrapping, i found a wonderful node module that wraps the github api, so im hopfully good. Ill take a look though.

**Comment by drazisil** [comment]
@adamlofting Took a look at gitribution-2, it's using the same node-github I am, just a lot cleaner code. I may 'borrow' the backend, just so I don't have to look at my horrid code. Thanks for mentioning it.
